test/capi/CMakeLists.txt

```diff
@@ -3,7 +3,7 @@ project(capi_test)
 make_boost_test(capi_parameters.cxx REQUIRES turi_capi)
 make_boost_test(capi_flexible_type.cxx REQUIRES turi_capi)
 make_boost_test(capi_sarray.cxx REQUIRES turi_capi)
-make_boost_test(capi_sframe.cxx REQUIRES turi_capi)
+make_boost_test(capi_sframe.cxx ../../src/unity/extensions/random_sframe_generation.cpp REQUIRES turi_capi)
```

Comment:
I think you can just use `make_boost_test(capi_sframe.cxx REQUIRES turi_capi random_sframe_generation)`  Note in src/unity/extensions/CMakeLists.txt, you can see how this file becomes a target there that these things can depend on.

Reply:
We tried that but it didn't work because random_sframe_generation is a dylib, so @zach recommended I add the random_sframe_generation.cpp file as a source file.

Reply:
@hoytak if you know of a better way to do this please share. I couldn't figure out how to link a unit test against a dylib extension.

Reply:
Okay, that's fine. 

Ideally, the sframe generation extensions mechanism in there shouldn't be done in that way.  We should actually pull these utilities into the main compile path.